lib/babe/epoch_handler.go

```diff
@@ -44,23 +44,36 @@ func newEpochHandler(epochNumber, firstSlot uint64, epochData *epochData, consta
 			epochData.threshold,
 			keypair,
 		)
-		if errors.Is(err, errOverPrimarySlotThreshold) {
+		if err == nil {
+			preRuntimeDigest, _ := types.NewBabePrimaryPreDigest(epochData.authorityIndex, i, proof.output, proof.proof).ToPreRuntimeDigest()
+			slotToPreRuntimeDigest[i] = preRuntimeDigest
+			logger.Debugf("epoch %d: claimed slot %d", epochNumber, i)
 			continue
-		} else if err != nil {
+		}
+		if !errors.Is(err, errOverPrimarySlotThreshold) {
+			return nil, fmt.Errorf("error running slot lottery at slot %d: %w", i, err)
+		}
```

Comment:
```suggestion
		if !errors.Is(err, errOverPrimarySlotThreshold) {
			continue
		}
```

this isn't an error, this causes us to return basically immediately